include/snitch/snitch_section.hpp

```diff
@@ -3,12 +3,21 @@
 
 #include "snitch/snitch_config.hpp"
 #include "snitch/snitch_test_data.hpp"
+#if SNITCH_WITH_TIMINGS
+#    include <chrono>
```

Comment:
We had, so far, managed to avoid including `<chrono>` in public headers. Could we keep it that way, to keep compilation time down? I just ran some tests, measuring GCC time to parse an empty `.cpp` with `#include`:
 - include `snitch_all.hpp` as of today's `main` branch: 38ms.
 - same, but add `<chrono>`: 46ms (+8ms, or +22%).

This feels like a high price to pay for just storing a time point. We could perhaps store the numerical value of the time point in the `struct` here, as an anonymous `std::size_t` or similar that is `>= sizeof(std::chrono::steady_clock::rep)`, and then convert back and forth to actual time points in the `*.cpp` files.

Reply:
Makes sense, I see that <type_traits> is included already so I use that instead to create a signed version of `size_t` (since not all platforms define ssize_t). I added some aliases in the snitch::impl for chrono in the cpp file as well to make the chrono code less verbose and avoid bugs if anyone wants to change clocks or duration types in the future.

Reply:
I ended up adding a new file `snitch_time.cpp` to wrap this up, since we needed chrono functionality elsewhere too (`registry`). In the end, all we need is a simple API with a function returning the "current time point" in some unspecified unit (I picked "number of nanoseconds since test run start" -- always positive with a `steady_clock` so a `std::size_t` is OK, and will only wrap around after ~600 years of execution, also OK) and another function to calculate the time difference between two time points.